compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -1372,16 +1372,22 @@ fn deny_equality_constraints(
                         if param.ident == *ident {
                             let param = ident;
                             match &full_path.segments[qself.position..] {
-                                [PathSegment { ident, .. }] => {
+                                [PathSegment { ident, args, .. }] => {
                                     // Make a new `Path` from `foo::Bar` to `Foo<Bar = RhsTy>`.
                                     let mut assoc_path = full_path.clone();
                                     // Remove `Bar` from `Foo::Bar`.
                                     assoc_path.segments.pop();
                                     let len = assoc_path.segments.len() - 1;
+                                    let gen_args = if let Some(gen_args) = args {
```

Comment:
```suggestion
                                    let gen_args = args.cloned();
```
?

Reply:
This didn't work, because 'PathSegment' uses `Option<P<GenericArgs>>`, whereas we only use `Option<GenericArgs>`. I settled for `let gen_args = args.as_ref().map(|p| (**p).clone());`, is that better?

Reply:
Yes, `map` is still better.